store/displayPreferences.ts

```diff
@@ -134,170 +122,100 @@ export const actions: ActionTree<
         throw new Error(
           'set display preferences status response = ' + response.status
         );
-      await dispatch('pushStateSuccess');
+      }
     } catch (error) {
-      await dispatch('pushStateFailure', { error });
+      const message = this.$i18n.t('failedSettingDisplayPreferences');
+      await dispatch(
+        'snackbar/pushSnackbarMessage',
+        { message, color: 'error' },
+        { root: true }
+      );
     }
+    commit('SYNCING_ENDED');
   },
-
-  /**
-   * Empty function for push state success in case we want to subscribe to it
-   */
-  // eslint-disable-next-line @typescript-eslint/no-empty-function
-  pushStateSuccess() {},
-
-  /**
-   * On push failure, logs a message
-   *
-   * @param {any} param0 - Vuex
-   * @param {any} param0.dispatch - Vuex dispatch
-   * @param {any} param1 - Payload
-   * @param {any} param1.error - Try-catch error
-   */
-  async pushStateFailure({ dispatch }, { error }: { error: unknown }) {
-    const message = this.$i18n.t('failedSettingDisplayPreferences');
-    await dispatch('requestError', { message, error });
-  },
-
-  /**
-   * Edits a custom preference and pushes it to the server
-   *
-   * @param {any} param0 - Vuex
-   * @param {any} param0.commit - Vuex commit
-   * @param {any} param0.dispatch - Vuex dispatch
-   * @param {any} param1 - Payload
-   * @param {string} param1.key - Key of custom pref to edit
-   * @param {string} param1.value - Value to apply
-   */
-  async editCustomPref(
-    { commit, dispatch },
-    { key, value }: { key: string; value: string }
-  ) {
-    commit('EDIT_CUSTOM_PREF', { pref: { key, value } });
+  async setDarkMode({ commit, dispatch }, { darkMode }: { darkMode: boolean }) {
+    commit('SET_DARK_MODE', { darkMode });
     if (this.$auth.loggedIn) await dispatch('pushState');
   },
-
-  /**
-   * Same as editCustomPref, but takes a boolean value and parses it to string so the serve can store it
-   *
-   * @param {any} param0 - Vuex
-   * @param {any} param0.dispatch - Vuex dispatch
-   * @param {any} param1 - Payload
-   * @param {string} param1.key - Key of custom pref to edit
-   * @param {boolean} param1.value - Value to apply
-   */
-  async editBooleanCustomPref(
-    { dispatch },
-    { key, value }: { key: string; value: boolean }
-  ) {
-    await dispatch('editCustomPref', { key, value: booleanToString(value) });
+  async setLocale({ commit, dispatch }, { locale }: { locale: string }) {
+    commit('SET_LOCALE', { locale });
+    if (this.$auth.loggedIn) await dispatch('pushState');
   },
-
   /**
    * Resets the state and reapply default theme
    *
-   * @param {any} param0 - Vuex
-   * @param {any} param0.commit - Vuex commit
-   * @param {any} param0.dispatch - Vuex dispatch
+   * @param {any} context - Vuex action context
+   * @param {any} context.commit - Vuex commit
+   * @param {any} context.dispatch - Vuex dispatch
    */
-  async resetState({ commit, dispatch }) {
+  resetState({ commit }) {
```

Comment:
Imo we shouldn't use JSDoc at all in Vuex stores because, as you mention, their TypeScript support is really bad and stuff like this happens. Didn't do that as @ThibaultNocchi spent probably quite some time writing it. 

I don't mind them though, so I will leave it up to him :).

Reply:
I do agree that typings are weird with Vuex now :p

Idk if there's a way to just keep the descriptions without the verbose stuff about the arguments?